play/src/front/Phaser/Game/GameScene.ts

```diff
@@ -2275,17 +2282,24 @@ ${escapedMessage}
         if (obj === undefined) {
             return undefined;
         }
-        return obj.value as string | number | boolean | undefined;
+
+        return z.union([z.string(), z.number(), z.boolean()]).optional().parse(obj.value);
```

Comment:
Not a big fan of this.
The type can be ANY type allowed for a type.

We should indeed improve the typing of "wa tiled" library.

Something like:

export type PossiblePropertyValue = string |number | boolean | JSON.Value | ...